custom_components/acwd/acwd_api.py

```
@@ -385,15 +387,16 @@ def get_usage_data(self, mode='B', date_from=None, date_to=None, str_date=None,
-            else:
-                _LOGGER.error("Unexpected response format")
-                return None
         except Exception as e:
-            _LOGGER.error(f"Error parsing usage data: {e}")
+            _LOGGER.error(f"Error decoding usage response: {e}")
+            return None
+
+        try:
+            usage_data = parse_api_response(result, endpoint="LoadWaterUsage")
+            _LOGGER.info("Retrieved usage data successfully")
+            return usage_data
```

Ruff (0.15.2)

[warning] Consider moving this statement to an `else` block

(TRY300)